src/provider/element-templates/util/templateUtil.js

```diff
@@ -16,23 +16,23 @@ export function unlinkTemplate(element, injector) {
 export function removeTemplate(element, injector) {
   const replace = injector.get('replace'),
         selection = injector.get('selection'),
-        modeling = injector.get('modeling');
+        bpmnFactory = injector.get('bpmnFactory');
 
   const businessObject = getBusinessObject(element);
 
   const type = businessObject.$type,
         eventDefinitionType = getEventDefinitionType(businessObject),
         label = getLabel(element);
 
+  const newBusinessObject = bpmnFactory.create(businessObject.$type);
```

Comment:
We need to be aware of different names here:

* Text annotaiton, group have different _names_

Reply:
Other than that I like this direction.

Reply:
Groups are not that straight forward, but should be all good now:
https://github.com/bpmn-io/bpmn-js-properties-panel/blob/905eaf8e2b82fcd9b20925ac20b6abe30184de8d/src/provider/element-templates/util/templateUtil.js#L112-L118